src/render.rs

```diff
@@ -0,0 +1,236 @@
+use matrix_sdk::{
+    events::room::{
+        encrypted::EncryptedEvent,
+        member::{MemberEvent, MembershipChange, MembershipState},
+        message::{
+            AudioMessageEventContent, EmoteMessageEventContent,
+            FileMessageEventContent, ImageMessageEventContent, MessageEvent,
+            MessageEventContent, NoticeMessageEventContent,
+            TextMessageEventContent, VideoMessageEventContent,
+        },
+    },
+    identifiers::UserId,
+};
+use weechat::Weechat;
+
+/// This trait describes events that can be rendered in the weechat UI
+pub(crate) trait RenderableEvent {
+    /// Convert the event into a string that will be displayed in the UI.
+    /// The displayname is taken as a parameter since it cannot be calculated from the event
+    /// context alone.
+    fn render(&self, displayname: &str) -> String;
+
+    /// Get the sender for this event this can be used to get the displayname.
+    fn sender(&self) -> &UserId;
+}
+
+impl RenderableEvent for EncryptedEvent {
+    // TODO: this is not implemented yet
+    fn render(&self, displayname: &str) -> String {
+        let color_user = Weechat::color("green"); // TODO: get per-user color
```

Comment:
Since we're going to use the user_id to get the color we probably can get away without storing the color so a simple `Weechat::info_get("nick_color_name", user_id)` will suffice here, but we'll need to put some power level indicator before the display name as well.

For example if someone has a power level of 50 or above we will want to represent this using the `@` symbol like under IRC for ops. That info will be present in the room member struct. So we will probably need to pass the whole struct to the render method instead of just the display name.

But we can leave this for another PR.